torch/_C/_distributed_c10d.pyi

```diff
@@ -294,7 +294,7 @@ class _DistributedBackendOptions:
 class Work:
     def is_completed(self) -> bool: ...
     def is_success(self) -> bool: ...
-    def exception(self) -> Any: ...
+    def exception(self) -> BaseException | None: ...
```

Comment:
Would `Exception | None` be better? Feels like all C++ side c10 exceptions are mapped into subclasses of `Exception` on Python side.